python/cudf/cudf/core/dataframe.py

```diff
@@ -5741,7 +5740,7 @@ def isin(self, values):
                 ) and isinstance(
                     values._column, cudf.core.column.CategoricalColumn
                 ):
-                    res = self._data[col] == values._column
+                    res = (self._data[col] == values._column).fillna(False)
```

Comment:
Should nulls not propagate through an `isin` operation?

Reply:
Yeah they should, your're right. Trouble is this is a place where we're testing pandas non-nullable types versus our nullable types. The test that was failing here will need a little adjusting. More to follow.

Reply:
Connected offline and raised https://github.com/rapidsai/cudf/issues/7556 to fix this in a follow up.

Reply:
Can you add `#TODO` comments with issue #7556?